osu.Game/Beatmaps/Drawables/DifficultyIcon.cs

```diff
@@ -58,11 +63,16 @@ public virtual Bindable<StarDifficulty> Current
         /// Creates a new <see cref="DifficultyIcon"/>. Will use provided beatmap's <see cref="BeatmapInfo.StarRating"/> for initial value.
         /// </summary>
         /// <param name="beatmap">The beatmap to be displayed in the tooltip, and to be used for the initial star rating value.</param>
+        /// <param name="mods">An array of mods to account for in the calculations</param>
         /// <param name="ruleset">An optional ruleset to be used for the icon display, in place of the beatmap's ruleset.</param>
-        public DifficultyIcon(IBeatmapInfo beatmap, IRulesetInfo? ruleset = null)
+        /// <param name="showTooltip">Whether to display a tooltip on hover. Defaults to false.</param>
+        public DifficultyIcon(IBeatmapInfo beatmap, IRulesetInfo? ruleset = null, Mod[]? mods = null, bool showTooltip = false)
```

Comment:
`ShowTooltip` already existed on this class as a public property. Now there is `ShowTooltip` (the public property) and `showTooltip` (the constructor argument), and the former actually decides whether to show the tooltip, but the latter is passed to `DifficultyIconTooltipContent`, wherein it is stored to `ShowTooltip`, and used *not* to determine whether to *show* itself, but rather to show the *extended* content.

Please help. This just cannot be.

Reply:
I've refactored the name of the variable & constructor to be more descriptive and more accurate. I've also updated the docstring for `DifficultyIcon` to accurate reflect what the variable is used for.

So that we're on the same page, there isn't an issue with the implementation here right? My intent is to have the parameter in `DifficultyIcon`'s constructor determine whether the extended details are shown or not. The rest is just passing that to `DifficultyIconTooltipContent` to be put to use. Also, are you suggesting that the private property should have a different name than the constructor argument?

Reply:
> there isn't an issue with the implementation here right? My intent is to have the parameter in `DifficultyIcon`'s constructor determine whether the extended details are shown or not

I guess not. I'm in two minds as to whether it should be a bool flag or a subclass, however.



> Also, are you suggesting that the private property should have a different name than the constructor argument?

Definitely yes.

Reply:
> I'm in two minds as to whether it should be a bool flag or a subclass, however.

That's also a good way to go about it. I wouldn't mind implementing a bool or subclass. Whichever is cleaner is probably best.

> Definitely yes.

Got it

Reply:
Decided to just copy the same implementation as `ShowTooltips` and make `ShowExtendedTooltip` a public field. 

It doesn't really feel right when the way for determining if the tooltip is visible is by modifying the public field, while the way for determining if the content of the tooltip is extended is by passing a constructor argument.

Reply:
I'm happy enough with the current implementation, so I'll mark this as resolved